Content.Server/Explosion/EntitySystems/ExplosionSystem.Processing.cs

```diff
@@ -264,35 +264,31 @@ public bool IsBlockingTurf(EntityUid uid, EntityQuery<PhysicsComponent> physicsQ
         var worldBox = spaceMatrix.TransformBox(gridBox);
         List<(EntityUid, TransformComponent)> list = new();
 
-        EntityUidQueryCallback callback = uid =>
+        void AddIntersecting(List<(EntityUid, TransformComponent)> list)
         {
-            if (processed.Contains(uid))
-                return;
-
-            var xform  = xformQuery.GetComponent(uid);
-
-            if (xform.ParentUid == lookup.Owner)
+            foreach (var uid in _entityLookup.GetEntitiesIntersecting(lookup, ref worldBox, LookupFlags.None))
             {
-                // parented directly to the map, use local position
-                if (gridBox.Contains(invSpaceMatrix.Transform(xform.LocalPosition)))
-                    list.Add((uid, xform));
+                if (processed.Contains(uid))
+                    return;
 
-                return;
-            }
+                var xform  = xformQuery.GetComponent(uid);
 
-            if (!metaQuery.TryGetComponent(uid, out var meta))
-                return;
+                if (xform.ParentUid == lookup.Owner)
+                {
+                    // parented directly to the map, use local position
+                    if (gridBox.Contains(invSpaceMatrix.Transform(xform.LocalPosition)))
+                        list.Add((uid, xform));
 
-            // Not parented to map. Likely in a container.
-            if (_containerSystem.IsEntityInContainer(uid, meta))
-                return;
+                    return;
+                }
 
-            // finally check if it intersects our tile
-            if (gridBox.Contains(invSpaceMatrix.Transform(xform.LocalPosition)))
-                list.Add((uid, xform));
-        };
+                // finally check if it intersects our tile
+                if (gridBox.Contains(invSpaceMatrix.Transform(xform.LocalPosition)))
```

Comment:
Like you pointed out, this should be WorldPosition. So I guess use transform system & pass in the xform query?